### PR TITLE
added controller for user

### PR DESCRIPTION
**Pull request by seanheinrichs**
Hey guys, before we write too much code I thought it could be a good time to discuss how we want to structure the backend. Here's what I was thinking for the route / controller combo.

Then I think what Mike made for services would be super valid as well for the place where the core reusable "business logic" goes. I'm open to any thoughts / feedback.

### CODE REVIEW

File: backend/src/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      userData = user.dataValues;
+
+      // Delete extraneous data
+      delete userData.createdAt;
+      delete userData.updatedAt;
+
+      res.send(userData);
+    })
+    .catch(err => console.log(err))
+};
```
Comment (seanheinrichs) [inline backend/src/controllers/user.js:26]: I'm changing this back to async await in next PR... woopsies

File: backend/src/routes/user.js [hunk 2]
```diff
@@ -1,85 +1,18 @@
...
-  res.send(userData);
-});
-
-router.get("/avatar/:userId", isAuth, attachCurrentUser, (req, res) => {
-  let filePath = path.join(__dirname, `../../data/${req.params.userId}.jpg`);
-  if (fs.existsSync(filePath)) res.sendFile(filePath);
-  else res.sendStatus(404);
-});
+const userController = require('../controllers/user');
+const { isAuth, attachCurrentUser } = require("../middleware");
```
Comment (mdesson) [inline backend/src/routes/user.js:7]: Would it be possible to move these below the library imports? It's a little weird to have local imports follow actual logic.

The way I was trying to organize it (loosely) was:

- library imports (express, etc.)
- local imports (our routes, models, middleware, etc.)
- Then the code and logic begins

Finishing your imports, then starting imports again is a little confusing to read

**Comment by seanheinrichs** [comment]
I mean, if we follow that logic to its absurd end, why not write the entire application in 1 file? The reason is obvious, it would be absolutely unreadable and unmaintainable. We have to decide where to draw the lines about what file should handle what code. Especially as the project grows to something larger than a couple files. Now it may not matter for us, or this project specifically because it is small, and we will be on the only ones working on it but it's good to start with solid practices now. 

The idea behind this is that the logic in our code should be readable, at a glance. If I want to see the routes I can hit for users, I shouldn't have to scroll through a 200-300 line file to see what's available, it serves me no purpose and for larger routes like squads this will become extremely cumbersome. Rather with this change, I can go to users, and can see all the routes available to me at a glance. The vast majority of the time, developers don't care about the implementation of another developer (unless it stops working, then it's our problem). In my opinion, there is a clear distinction between what the routes are, and what they do. This seems like a logical line to draw because again, the implementation of the routes can be a black box. 99% of the time we don't care. If I am interested in how a specific route works, I can then go into the controller to find out. 

Again, writing maintainable code isn't JUST about reducing the amount of lines you write, its about organizing the logic in a manner that makes sense. I think you're comment about the middleware is interesting though, we could move those into the controller file as well...

**Comment by mdesson** [comment]
Sold! This makes perfect sense, like C++ header files. You just need to see what's available at a glance.

**Comment by seanheinrichs** [comment]
ah yeah that's actually a great comparison!

**Pull request merged**